system/jlib/jsocket.cpp

```diff
@@ -3931,11 +3941,16 @@ class CSocketSelectThread: public CSocketBaseThread
             T_SOCKET sock = popfds(s);
             if (!sock)
                 break;
-            if (sock!=dummysock) {
-                SelectItem si = findhash(sock); // nb copies
-                if (!si.del) {
-                    si.mode = mode;
-                    tonotify.append(si);
+            if (sock!=dummysock)
+            {
+                SelectItem &si = findhash(sock);
+                if (!si.del)
+                {
+                    tonotify.append(si); // nb copies
+                    SelectItem &itm = tonotify.element(tonotify.length()-1);
```

Comment:
It's a bit odd the tonotify array is a copy array (allocates new SelectItem's and copies content of SelectItem structs appended to it).
So it could take a copy and then modify that, but the append would then make another copy.
Practically it makes no difference and avoids a copy, so I think it's okay.